packages/aws-rfdk/lib/lambdas/nodejs/ecr-provider/handler.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-console */
+
+import * as path from 'path';
+
+import { LambdaContext } from '../lib/aws-lambda';
+import { CfnRequestEvent, SimpleCustomResource } from '../lib/custom-resource';
+import {
+  ThinkboxEcrProvider,
+} from '../lib/ecr-provider';
+
+/**
+ * The input to this Custom Resource
+ */
+export interface ThinkboxEcrProviderResourceProperties {
+  /**
+   * The desired ECR region. If not provided, the global ECR is used.
+   */
+  readonly Region?: string;
+
+  /**
+   * A random string that forces the Lambda to run again and obtain the latest ECR.
+   */
+  readonly ForceRun?: string;
+}
+
+/**
+ * Output of this Custom Resource.
+ */
+export interface ThinkboxEcrProviderResourceOutput {
+  /**
+   * The ARN prefix of the ECR repository. This can be suffixed with the recipe name to get the final ECR repository
+   * ARN.
+   */
+  readonly EcrArnPrefix: string;
+}
+
+/**
+ * This custom resource will parse and return the base ECR ARN or URI containing Thinkbox published Docker Images. A
+ * region can be specified to get a regional ECR base ARN. Otherwise, the global ECR base URI is returned.
+ */
+export class ThinkboxEcrProviderResource extends SimpleCustomResource {
+  readonly ecrProvider: ThinkboxEcrProvider;
+
+  constructor() {
+    super();
+    this.ecrProvider = new ThinkboxEcrProvider(path.join(
+      __dirname,
+      '..',
+      'lib',
+      'ecr-provider',
+      'mock-data.json',
+    ));
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public validateInput(data: object): boolean {
+    return this.isEcrProviderResourceProperties(data);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public async doCreate(_physicalId: string, resourceProperties: ThinkboxEcrProviderResourceProperties): Promise<ThinkboxEcrProviderResourceOutput> {
+    let result: any;
+    if (resourceProperties.Region) {
+      result = {
+        EcrArnPrefix: await this.ecrProvider.getRegionalEcrBaseArn(resourceProperties.Region),
```

Comment:
Are the AWS Thinkbox containers actually being published to regional ECRs? Or just the global public ECR?

Reply:
I've removed all regional ECR code since we're just using ECR public now

Reply:
Confirmed regional code is removed - resolved